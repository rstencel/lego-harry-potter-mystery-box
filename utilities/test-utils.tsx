import React, { FC, ReactElement } from "react"
import { render, RenderOptions } from "@testing-library/react"
import { IntlProvider } from "react-intl"
import messages from "../translations/en"

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <IntlProvider locale="en" defaultLocale="en" messages={messages}>
    {children}
  </IntlProvider>
)

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options })

export { customRender as render }