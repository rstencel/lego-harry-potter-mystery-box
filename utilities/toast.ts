import { toast, TypeOptions } from 'react-toastify'

export const showToast = (message: string, type: TypeOptions, onCloseCallback?: () => void) => {
  toast.onChange(payload => {
    if (payload.status === "removed" && onCloseCallback) {
      onCloseCallback()
    }
  })
  
  toast(message, {
    type,
    position: "top-center",
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: false,
    pauseOnHover: false,
    draggable: false,
    progress: undefined,
    theme: "light"
  })
}