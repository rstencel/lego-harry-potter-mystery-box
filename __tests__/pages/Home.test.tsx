import Home from "@/pages/index"
import { render } from "../../utilities/test-utils"
import fetchMock from "jest-fetch-mock"

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn()
  })
}))

beforeEach(() => {
  fetchMock.doMock()
})

describe("Home", () => {
  it("Should match snapshot", () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })
})