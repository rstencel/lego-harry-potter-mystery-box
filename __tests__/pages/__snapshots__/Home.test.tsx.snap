// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should match snapshot 1`] = `
<div>
  <main>
    <h1>
      LEGO minifigs mystery box
    </h1>
    <button
      style="margin-bottom: 2rem;"
    >
      Let's go!
    </button>
    <div
      class="loader-with-message"
      style="visibility: hidden;"
    >
      <div
        class="loader"
      />
      <p
        style="color: white; margin-left: 1rem;"
      >
        Drawing minifigs ...
      </p>
    </div>
  </main>
</div>
`;
